src/viking.js

```diff
@@ -1,11 +1,65 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength){
+        this.health = health
+        this.strength = strength
+    }
+    attack(){
+        return this.strength
+    }
+    receiveDamage(damage){
+        this.health = this.health - damage;
+}
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor(name, health, strength){
+        super(health, strength)
+
+        this.name = name;
+    }
+    receiveDamage(damage){
+        this.health -= damage;
+        if (this.health > damage){
```

Comment:
Here we should check if the viking survived after receiving the damage
```suggestion
        if (this.health > 0){
```